pkg/ambassador/tap/conduit/conduit.go

```diff
@@ -137,9 +144,12 @@ func (c *Conduit) Connect(ctx context.Context) error {
 // Disconnect closes the connection from NSM, closes all streams
 // and stop the VIP watcher
 func (c *Conduit) Disconnect(ctx context.Context) error {
+	c.isDisconnecting.Store(true)
```

Comment:
Move the `c.mu.Lock()` a few lines up and drop the unnecessary extra ctxMu lock.